### PR TITLE
Split conn tests into chunks

### PR DESCRIPTION
**Pull request by viktor-kurchenko**
New functions implemented for the test builder component to split connectivity tests into a few chunks:
1. Network performance tests
2. Conn disrupt tests
3. Concurrent tests
4. Sequential tests
5. Final tests

This refactoring is a preparation for the conn tests concurrent run.

### CODE REVIEW

File: connectivity/builder/builder.go [hunk 1]
```diff
@@ -73,8 +73,29 @@ var (
...
+func ConnDisruptTests(ct *check.ConnectivityTest) error {
+	tests := []testBuilder{
+		noInterruptedConnections{},
+		noIpsecXfrmErrors{},
+	}
+	return getTests(ct, tests)
+}
+
+// ConcurrentTests returns a slice with all connectivity tests that can be run concurrently.
+func ConcurrentTests(ct *check.ConnectivityTest) error {
```
Comment (michi-covalent) [inline connectivity/builder/builder.go:97]: how did you determine which tests can run concurrently? for example, i'm not sure how `noPolicies` can run concurrently with `allowAllExceptWorld` 💭

Comment (viktor-kurchenko) [reply connectivity/builder/builder.go:97]: Concurrent tests will run in separate namespaces.
Probably I need to mention this somewhere...

Comment (michi-covalent) [reply connectivity/builder/builder.go:97]: ah ok sounds good, yeah that wasn't obvious to me from this pull request.

Comment (viktor-kurchenko) [reply connectivity/builder/builder.go:98]: I've improved function description and PR comment. Thanks!

File: connectivity/builder/builder.go [hunk 2]
```diff
@@ -73,8 +73,30 @@ var (
...
+func ConnDisruptTests(ct *check.ConnectivityTest) error {
+	tests := []testBuilder{
+		noInterruptedConnections{},
+		noIpsecXfrmErrors{},
+	}
+	return getTests(ct, tests)
+}
+
+// ConcurrentTests returns a slice with all connectivity tests that can be run concurrently.
+// Each test should be run in a separate namespace.
```
[the remark this answers is not in the document]
Comment (viktor-kurchenko) [reply connectivity/builder/builder.go:97]: This is up to the caller. I'm going to do this in a follow up PR.

File: connectivity/builder/builder.go [hunk 3]
```diff
@@ -73,8 +73,30 @@ var (
...
 	clientLabel  = map[string]string{"name": "client"}
 	client2Label = map[string]string{"name": "client2"}
+)
+
+type testBuilder interface {
+	build(ct *check.ConnectivityTest, templates map[string]string)
+}
 
-	primaryTests = []testBuilder{
+// NetworkPerformanceTests returns a slice with network performance connectivity tests.
```
[the remark this answers is not in the document]
Comment (viktor-kurchenko) [reply connectivity/builder/builder.go]: Fixed, thank you!

**Review by tklauser — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]

**Pull request merged**